lib/pengine/tests/utils/pe__sort_node_uname_test.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/common/unittest_internal.h>
+#include <crm/pengine/internal.h>
+
+struct pe_node_shared_s node1_details;
+struct pe_node_shared_s node2_details;
+
+pe_node_t node1 = {.details = &node1_details};
+pe_node_t node2 = {.details = &node2_details};
+pe_node_t bad_node1 = {.details = NULL};
+pe_node_t bad_node2 = {.details = NULL};
+
+static void
+nodes_equal(void **state)
+{
+    node1.details->uname = "a";
+    node2.details->uname = "a";
+
+    // NULL tests
+    assert_int_equal(pe__sort_node_uname(NULL, NULL), 0);
+    assert_int_equal(pe__sort_node_uname(&bad_node1, NULL), 0);
+    assert_int_equal(pe__sort_node_uname(NULL, &bad_node2), 0);
+    assert_int_equal(pe__sort_node_uname(&bad_node1, &bad_node2), 0);
+
+    // Compare unames
+    assert_int_equal(pe__sort_node_uname(&node1, &node2), 0);
+}
+
+static void
+node1_first(void **state)
+{
+    node1.details->uname = "a";
+    node2.details->uname = "b";
+
+    // NULL tests
+    assert_int_equal(pe__sort_node_uname(NULL, &node2), -1);
+    assert_int_equal(pe__sort_node_uname(&bad_node1, &node2), -1);
+
+    // Compare unames
+    assert_int_equal(pe__sort_node_uname(&node1, &node2), -1);
+}
+
+static void
+node2_first(void **state)
+{
+    node1.details->uname = "b";
+    node2.details->uname = "a";
+
+    // NULL tests
+    assert_int_equal(pe__sort_node_uname(&node1, NULL), 1);
+    assert_int_equal(pe__sort_node_uname(&node1, &bad_node2), 1);
+
+    // Compare unames
+    assert_int_equal(pe__sort_node_uname(&node1, &node2), 1);
+}
+
+PCMK__UNIT_TEST(NULL, NULL,
+                cmocka_unit_test(nodes_equal),
+                cmocka_unit_test(node1_first),
+                cmocka_unit_test(node2_first))
```

Comment:
probably worth testing something like node10 vs node1

Reply:
We definitely can for sanity. I thought that might be a double-test of `pcmk__numeric_strcasecmp()`, similar to what came up in another unit test PR recently.

Reply:
Yep normally I wouldn't consider it important, but that is semantically essential to comparing node names, so it makes sure if someone changes the implementation they don't introduce a regression. By comparison we don't need to test that nodeA < nodeB, since that is a generic string comparison and not specific to node names.

Reply:
replaced "a" vs "b" with "node9" vs "node10"